### ISSUE TITLE
No warning given for some typedef mismatches

### ISSUE THREAD

**Issue by supersteves** [google/closure-compiler#1149]
Externs [as advised](https://github.com/google/closure-compiler/pull/869#issuecomment-77226492):
``` javascript
/** @type {Object} */
var myUnusedNamespaceToPreventRenaming;
myUnusedNamespaceToPreventRenaming.something;

/** @typedef {{ something: string, another: string }} */
var TEST;
```

Valid code that correctly emits no error:
``` javascript
/** @param {TEST} pure  */
var myFunc = function(pure) {};

myFunc({ something: "x", another: "x" });
```

Code with a typo that **correctly emits an error**:
``` javascript
/** @param {TEST} pure  */
var myFunc = function(pure) {};

myFunc({ something: "x", anotherBAD: "x" });
```
```
input0:4: WARNING - actual parameter 1 of myFunc does not match formal parameter
found   : {another: (string|undefined), anotherBAD: string, something: ?}
required: {another: string, something: string}
myFunc({ something: "x", anotherBAD: "x" });
       ^
```

Code with an apparently equivalent typo that **fails to emit an error**:
``` javascript
/** @param {TEST} pure  */
var myFunc = function(pure) {};

myFunc({ somethingBAD: "x", another: "x" });
```

https://closure-compiler-debugger.appspot.com/#input0%3D%252F**%2520%2540param%2520%257BTEST%257D%2520pure%2520%2520*%252F%250Avar%2520myFunc%2520%253D%2520function(pure)%2520%257B%257D%253B%250A%250AmyFunc(%257B%2520somethingBAD%253A%2520%2522x%2522%252C%2520another%253A%2520%2522x%2522%2520%257D)%253B%26input1%26conformanceConfig%26externs%3D%252F**%2520%2540type%2520%257BObject%257D%2520*%252F%250Avar%2520myUnusedNamespaceToPreventRenaming%253B%250AmyUnusedNamespaceToPreventRenaming.something%253B%250A%250A%252F**%2520%2540typedef%2520%257B%257B%2520something%253A%2520string%252C%2520another%253A%2520string%2520%257D%257D%2520*%252F%250Avar%2520TEST%253B%250A%26refasterjs-template%26CHECK_SYMBOLS%3D1%26CHECK_TYPES%3D1%26CLOSURE_PASS%3D1%26MISSING_PROPERTIES%3D1

Weird. Is my approach to prevent renaming just plain bad and confusing the compiler, or should this work? (note I'm exploring alternatives to this renaming approach anyway)

**Comment by supersteves** [comment]
This is another case (like https://github.com/google/closure-compiler/issues/1150) where you'd expect a warning or compiler crash but instead it silently passes.

**Comment by supersteves** [comment]
Is there a place or mechanism in the codebase for creating unit(ish) tests for scenarios like this? When I find a problem and am able to reproduce it in the debugger, I'd like to be able to contribute it in the form of a unit test that could be added when the bug was resolved. For example, a unit test that automatically parses the debugger URL and has expected or not expected substring to look for in the jserrors.

**Comment by supersteves** [comment]
Update: this is an OTI limitation; NTI handles it properly:
https://closure-compiler-debugger.appspot.com/#input0%3D%252F**%2520%2540param%2520%257BTEST%257D%2520pure%2520%2520*%252F%250Avar%2520myFunc%2520%253D%2520function(pure)%2520%257B%257D%253B%250A%250AmyFunc(%257B%2520somethingBAD%253A%2520%2522x%2522%252C%2520another%253A%2520%2522x%2522%2520%257D)%253B%26input1%26conformanceConfig%26externs%3D%252F**%2520%2540type%2520%257BObject%257D%2520*%252F%250Avar%2520myUnusedNamespaceToPreventRenaming%253B%250AmyUnusedNamespaceToPreventRenaming.something%253B%250A%250A%252F**%2520%2540typedef%2520%257B%257B%2520something%253A%2520string%252C%2520another%253A%2520string%2520%257D%257D%2520*%252F%250Avar%2520TEST%253B%250A%26refasterjs-template%26includeDefaultExterns%3D1%26CHECK_SYMBOLS%3D1%26CHECK_TYPES%3D1%26CHECK_TYPES_NEW_INFERENCE%3D1%26CLOSURE_PASS%3D1%26MISSING_PROPERTIES%3D1
